File: Level1/두 개 뽑아서 더하기/Solution.java
import java.util.Arrays;
import java.util.HashSet;
import java.util.Iterator;

class Solution {
	public static int[] solution(int[] numbers) {
		HashSet<Integer> set = new HashSet<>();
		
		int size = numbers.length;
		for(int i=0; i<size; i++) {
			for(int j=0; j<size; j++) {
				if(i != j)
					set.add(numbers[i] + numbers[j]);
			}
		}
		
		
		int[] ans = new int[set.size()];
		
		Iterator<Integer> iter = set.iterator();
		
		int i = 0;
		while(iter.hasNext()) {
			int a = iter.next();
			ans[i++] = a;
		}
		Arrays.sort(ans);
		return ans;
	}
}

File: Level1/두 개 뽑아서 더하기/solution.cpp
#include <string>
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> numbers) {
    vector<int> ans;

    for (int i = 0; i < numbers.size(); i++) {
        for (int j = i + 1; j < numbers.size(); j++) {
            ans.push_back(numbers[i] + numbers[j]);
        }
    }

    sort(ans.begin(), ans.end());
    ans.erase(unique(ans.begin(), ans.end()), ans.end());
    return ans;
}

int main() {
    vector<int> numbers = { 2,1,3,4,1 };
    vector<int> ans = solution(numbers);

    for (int a : ans) {
        cout << a << endl;
    }

    return 0;
}